src/main/kotlin/org/rust/ide/refactoring/RsImportOptimizer.kt

```diff
@@ -50,16 +50,21 @@ class RsImportOptimizer : ImportOptimizer {
 
     companion object {
 
-        private fun optimizeUseSpeck(psiFactory: RsPsiFactory, useSpeck: RsUseSpeck) {
-            if (removeCurlyBraces(psiFactory, useSpeck)) return
-            val useSpeckList = useSpeck.useGroup?.useSpeckList ?: return
-            if (useSpeckList.size < 2) return
-            useSpeckList.forEach { optimizeUseSpeck(psiFactory, it) }
-            val sortedList = useSpeckList
+        /** Returns false if use speck is empty and should be removed */
```

Comment:
These booleans look tricky. `removeCurlyBraces` returns `true` if braces were deleted successfully. `optimizeUseSpeck` returns `true` if `useSpeck` is OK and should not be removed. Probably the logic can be changed somehow to become more clear and simple

Reply:
Can we think of them as being consistent in the sense that they return `true` if they successfully did what they are supposed to do?) Also I added a comment to both `removeCurlyBraces` and `removeUseSpeckIfEmpty`

Reply:
OK, the comments help to understand the logic. Also, I'd rename `removeCurlyBraces` to `removeCurlyBracesIfTrivial` or `removeCurlyBracesIfPossible` since `removeUseSpeck` was renamed to `removeUseSpeckIfEmpty`